#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *
from os.path import abspath, basename, dirname, splitext

PACKAGE = basename(dirname(dirname(abspath(__file__))))

gen = ParameterGenerator()

gen.add("control_rate", double_t, 0, default=10.0, min=1.0, max=1000.0, description="Control rate in Hz")
gen.add("x_y_tolerance_increase", double_t, 0, default=1.0, min=0.0, max=10.0,
        description="Scaling of x and y tolerance")
gen.add("yaw_tolerance_increase", double_t, 0, default=1.0, min=0.0, max=10.0, description="Scaling of yaw tolerance")
gen.add("transformation_timeout", double_t, 0, default=0.1, min=0.0, max=10.0,
        description="Timeout for waiting for transforms")
gen.add("x_y_close_to_goal_tolerance", double_t, 0, default=1.0, min=0.0, max=10.0,
        description="Absolute values of x and y tolerance for the close to goal detection")
gen.add("yaw_close_to_goal_tolerance", double_t, 0, default=1.5708, min=0.0, max=3.1416,
        description="Absolute values of yaw tolerance for the close to goal detection")

exit(gen.generate(PACKAGE, PACKAGE + "_node", splitext(basename(__file__))[0]))
